volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -33,8 +37,30 @@ def get_requirements(cls):
             ),
         ]
 
-    def _generator(self):
-        layer = self.context.layers[self.config["primary"]]
+    @staticmethod
+    def enumerate_mft_records(
+        context: interfaces.context.ContextInterface,
+        config: interfaces.configuration.HierarchicalDict,
+        config_path: str,
+        attr_callback,
+    ) -> interfaces.objects.ObjectInterface:
+        try:
+            primary = context.layers[config["primary"]]
```

Comment:
I'd pass in `primary_layer_name`  as a string.  If this gets called from a different plugin the config may not match exactly and then they'll have to construct a fake one just to get to that value.

Reply:
done